RecoHGCal/TICL/plugins/TrackstersMergeProducer.cc

```diff
@@ -528,9 +539,32 @@ void TrackstersMergeProducer::produce(edm::Event &evt, const edm::EventSetup &es
     }
   }
 
+  // Compute timing
+  computeTime(*resultCandidates);
+
   evt.put(std::move(resultCandidates));
 }
 
+void TrackstersMergeProducer::computeTime(std::vector<TICLCandidate> &resultCandidates) {
+  for (auto &cand : resultCandidates) {
+    auto time = 0.;
+    auto timeErr = -1.;
+    for (const auto &tr : cand.tracksters()) {
+      if (tr->timeError() > 0) {
+        auto invTimeESq = pow(tr->timeError(), -2);
+        time += tr->time() * invTimeESq;
+        timeErr += invTimeESq;
+      }
+    }
+    if (timeErr > 0) {
+      timeErr = 1. / timeErr;
+
+      cand.setTime(time * timeErr);
+      cand.setTimeError(sqrt(timeErr));
```

Comment:
* If a `candidate` has no `trackster`, the `if` condition at line 559 is never `true`, so `timeError=-1.f` and the `sqrt(timeErr)` will fail.

* Not sure, also, about the logic of line 556: are you incrementing `timeErr` starting from `-1.f`?

* Can you attach the time of the `trackster` passed via the `addTrackster` method of the `candidate`? In this way, you can limit this final loop to the cases in which more than one `trackster` has been assigned to the `candidate`.

Reply:
Line 556 was a typo, should be `0.f`
All fixed in the last commit.